src/Corax/Utils/SpatialOptions.cs

```diff
@@ -0,0 +1,142 @@
+        Strategy = SpatialSearchStrategy.GeohashPrefixTree;
+        MaxTreeLevel = DefaultGeohashLevel;
+        MinX = -180;
+        MaxX = 180;
+        MinY = -90;
+        MaxY = 90;
+        Units = SpatialUnits.Kilometers;
+    }
+
+    internal SpatialOptions(SpatialOptions options)
+    {
+        Type = options.Type;
+        Strategy = options.Strategy;
+        MaxTreeLevel = options.MaxTreeLevel;
+        MinX = options.MinX;
+        MaxX = options.MaxX;
+        MinY = options.MinY;
+        MaxY = options.MaxY;
+        Units = options.Units;
+    }
+
+    public SpatialFieldType Type { get; set; }
+    public SpatialSearchStrategy Strategy { get; set; }
+    public int MaxTreeLevel { get; set; }
+    public double MinX { get; set; }
+    public double MaxX { get; set; }
+    public double MinY { get; set; }
+    public double MaxY { get; set; }
+
+    /// <summary>
+    /// Circle radius units, only used for geography indexes
+    /// </summary>
+    public SpatialUnits Units { get; set; }
+
+    protected bool Equals(SpatialOptions other)
+    {
+        var result = Type == other.Type && Strategy == other.Strategy;
+
+        if (Type == SpatialFieldType.Geography)
+        {
+            result = result && Units == other.Units;
+        }
+
+        if (Strategy != SpatialSearchStrategy.BoundingBox)
+        {
+            result = result && MaxTreeLevel == other.MaxTreeLevel;
+
+            if (Type == SpatialFieldType.Cartesian)
+            {
+                result = result
+                         && MinX.Equals(other.MinX)
+                         && MaxX.Equals(other.MaxX)
+                         && MinY.Equals(other.MinY)
+                         && MaxY.Equals(other.MaxY);
+            }
+        }
+
+        return result;
+    }
+
+    public override bool Equals(object obj)
+    {
+        if (ReferenceEquals(null, obj))
+            return false;
+        if (ReferenceEquals(this, obj))
+            return true;
+        if (obj.GetType() != GetType())
+            return false;
+        return Equals((SpatialOptions)obj);
+    }
+
+    public override int GetHashCode()
+    {
+        unchecked
+        {
+            int hashCode = (int)Type;
+            hashCode = (hashCode * 397) ^ (int)Strategy;
+
+            if (Type == SpatialFieldType.Geography)
+            {
+                hashCode = (hashCode * 397) ^ Units.GetHashCode();
+            }
+
+            if (Strategy != SpatialSearchStrategy.BoundingBox)
+            {
+                hashCode = (hashCode * 397) ^ MaxTreeLevel;
+
+                if (Type == SpatialFieldType.Cartesian)
+                {
+                    hashCode = (hashCode * 397) ^ MinX.GetHashCode();
+                    hashCode = (hashCode * 397) ^ MaxX.GetHashCode();
+                    hashCode = (hashCode * 397) ^ MinY.GetHashCode();
+                    hashCode = (hashCode * 397) ^ MaxY.GetHashCode();
+                }
+            }
+
```

Comment:
Use the `Hashing.HashCombiner.Combine` method BUT it is very important that what you put there is a hash. If it is not all bets are off. Line 92 is an example of a no no...

The best way to calculate a proper hash is to put in a `stackalloc` using `Unsafe.Write` all the primitive types and just execute a `Marvin32.CalculateInline()` over the content. Also dont request hash codes of primitive types.

Reply:
Oh, this is copied code from Server for making querying more convenient for now but good to know. I will rebuild that :)